### ISSUE TITLE
Inputfiles not copied to WN for GaudiExec

### ISSUE THREAD

**Issue by mesmith75** [ganga-devs/ganga#783]
I think this is just a GaudiExec issue. The files listed under ```j.inputfiles = ['somefile.py']``` are not being copied over to the worker node. They are put into the tar in the input dir ```_input_sandbox_13_master.tgz ``` but I can't find that on the WN.

**Comment by rob-c** [comment]
OK, which backend is this for?
(I should really get the gangamon server running again and restore the report functionality)
I'm curious to see the full trace from a job as this should be extracted at the start of the job running and I think I left the extract process fairly verbose to know what is extracted where.

If this is on DIRAC it's possible that the file is not extracted due to it already being inside one of the many levels of tar files being used in the process of getting files to the WN.
If DIRAC could you point me to an example job so I can have a look at the sandboxes.

**Comment by mesmith75** [comment]
@rob-c This is on Dirac. The job setup is very generic:
```python
 j = Job(name="test", application = GaudiExec(), backend = Dirac())
j.application.directory = '/afs/cern.ch/work/m/masmith/GangaTest/test/DaVinciDev_v40r1'
j.application.options =  ['cocktail_options.py']
j.inputfiles = ['tuplebranchinfos.py']
j.submit()
```
where the tuplebranchinfos.py is just generic python file. I'll look a bit further if I can find some time.

**Comment by rob-c** [comment]
OK, I'll try and submit a test job tomorrow and have a look at the job logging as a workaround may be as simple as calling multiple decompression runs over the files on the WN.

**Comment by rob-c** [comment]
(Or at least re-building the index after extracting each file and doing a better job of remembering what files we've already de-compressed, should be possible with some simple set logic)

**Comment by egede** [comment]
We should be careful in allowing a user to submit a compressed tarball as an inputfile (that then should stay compressed).

**Comment by rob-c** [comment]
@egede This gets difficult as if the user requests it as a `DiracFile` there is no way of stopping Dirac's own auto-decompression on the file.
We could try to have a list of files we want to decompress and in what order but this seems fragile. @mesmith75 Can you provide a jobID in this case to help with the debugging?

**Comment by mesmith75** [comment]
@rob-c a jobID is 139522694 .

**Comment by rmatev** [comment]
Any news here? I am also affected by this bug.

**Comment by mesmith75** [comment]
I have created a PR for a fix #815 . I can't quite follow the logic of how the input sandbox is prepared though so it is best that @rob-c checks it. This should go into 6.2.1 which I hope to start the release for on Monday evening.

**Comment by mesmith75** [comment]
Resolved by #822
